Panic in v0.9.*

**Describe the bug**
When the tailscale API returns additional error data, the plugin will crash due to not setting the summary field in the diagnostic error.

Reported in https://github.com/davidsbond/terraform-provider-tailscale/issues/89

Example

```

panic: Invalid diagnostic: empty summary. This is always a bug in the provider implementation

goroutine 55 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/internal/plugin/convert.DiagsToProto({0x14000096300, 0x2, 0x2})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/internal/plugin/convert/diagnostics.go:72 +0x2d4
github.com/hashicorp/terraform-plugin-sdk/v2/internal/plugin/convert.AppendProtoDiag({0x0, 0x0, 0x0}, {0x10169fbe0, 0x1400000c1f8})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/internal/plugin/convert/diagnostics.go:25 +0x70
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400030c078, {0x1017854b8, 0x14000050340}, 0x14000316320)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.1/helper/schema/grpc_provider.go:978 +0xe90
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400033c680, {0x101785560, 0x14000112810}, 0x140004fc1c0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x338
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x101741b40, 0x1400033c680}, {0x101785560, 0x14000112810}, 0x14000300360, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000328700, {0x101790af8, 0x140004024e0}, 0x1400000a360, 0x140003ca540, 0x101c127a0, 0x0)
        google.golang.org/grpc@v1.43.0/server.go:1282 +0xc60
google.golang.org/grpc.(*Server).handleStream(0x14000328700, {0x101790af8, 0x140004024e0}, 0x1400000a360, 0x0)
        google.golang.org/grpc@v1.43.0/server.go:1616 +0xa34
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400042c520, 0x14000328700, {0x101790af8, 0x140004024e0}, 0x1400000a360)
        google.golang.org/grpc@v1.43.0/server.go:921 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.43.0/server.go:919 +0x1f0

```

--- a/tailscale/provider.go
+++ b/tailscale/provider.go
@@ -87,7 +87,7 @@ func diagnosticsError(err error, message string, args ...interface{}) diag.Diagn
 			for _, e := range dt.Errors {
 				diags = append(diags, diag.Diagnostic{
 					Severity: diag.Error,
-					Detail:   fmt.Sprintf("user: %s\nerror: %s", dt.User, e),
+					Summary:  fmt.Sprintf("user: %s\nerror: %s", dt.User, e),
 				})
 			}
 		}

